src/java.base/share/classes/sun/security/jca/ProviderList.java

```diff
@@ -530,13 +558,23 @@ private Service tryGet(int index) {
                     // simple lookup
                     Service s = p.getService(type, algorithm);
                     if (s != null) {
+                        if (!RestrictedSecurity.isServiceAllowed(s)) {
+                            // We're in restricted security mode which does
+                            // not allow this service, skip it.
+                            continue;
+                        }
                         addService(s);
                     }
```

Comment:
I think this is clearer if we avoid the negation here; suggest:
```java
                    if ((s != null) && RestrictedSecurity.isServiceAllowed(s)) {
                        addService(s);
                    }
```
Like for line 572... below.